src/app/panels/text-panel/text-panel.component.ts

```diff
@@ -87,17 +98,29 @@ export class TextPanelComponent implements OnInit, OnDestroy {
   }
 
   toggleSecondaryContent(newContent: string) {
+    this.msDescOpen = !this.msDescOpen;
```

Comment:
The logic you put into this function is completely wrong. Try to open and close a secondary content (e.g. info) and see what happens.